src/main/java/at/hannibal2/skyhanni/features/chat/PowderMiningChatFilter.kt

```diff
@@ -0,0 +1,376 @@
+                val amountParsed = amountStr.toInt()
+                return if (amountParsed < config.essenceFilterThreshold) "powder_mining_essence"
+                else "no_filter"
+            }
+        }
+
+        //All the 'simple' boolean rewards
+        val rewardPatterns = mapOf(
+            ascensionRopeRewardPattern to ASCENSION_ROPE to "powder_mining_ascension_rope",
+            wishingCompassRewardPattern to WISHING_COMPASS to "powder_mining_wishing_compass",
+            oilBarrelRewardPattern to OIL_BARREL to "powder_mining_oil_barrel",
+            prehistoricEggPattern to PREHISTORIC_EGG to "powder_mining_prehistoric_egg",
+            pickonimbusPattern to PICKONIMBUS to "powder_mining_pickonimbus",
+            jungleHeartPattern to JUNGLE_HEART to "powder_mining_jungle_heart",
+            sludgeJuicePattern to SLUDGE_JUICE to "powder_mining_sludge_juice",
+            yoggiePattern to YOGGIE to "powder_mining_yoggie",
+            robotPartsPattern to ROBOT_PARTS to "powder_mining_robot_parts",
+            treasuritePattern to TREASURITE to "powder_mining_treasurite",
+        )
+        for ((patternToReward, returnReason) in rewardPatterns) {
+            if (patternToReward.first.matches(ssMessage)) {
+                return if (config.simplePowderMiningTypes.contains(patternToReward.second)) returnReason
+                else "no_filter"
+            }
+        }
+
+        //Goblin Eggs
+        goblinEggPattern.matchMatcher(ssMessage) {
+            if (config.goblinEggs != PowderMiningFilterConfig.GoblinEggFilterEntry.SHOW_ALL) {
+                if (config.goblinEggs == PowderMiningFilterConfig.GoblinEggFilterEntry.HIDE_ALL) return "powder_mining_goblin_eggs"
+
+                val colorStr = groupOrNull("color")?.lowercase() ?: ""
+                when (colorStr) {
+                    //'Colorless', base goblin eggs will never be shown in this code path
+                    "" -> return "powder_mining_goblin_eggs"
+                    "green" -> return if (config.goblinEggs > PowderMiningFilterConfig.GoblinEggFilterEntry.GREEN_UP) {
+                        return "powder_mining_goblin_eggs"
+                    } else "no_filter"
+                    "yellow" -> return if (config.goblinEggs > PowderMiningFilterConfig.GoblinEggFilterEntry.YELLOW_UP) {
+                        "powder_mining_goblin_eggs"
+                    } else "no_filter"
+                    "red" -> return if (config.goblinEggs > PowderMiningFilterConfig.GoblinEggFilterEntry.RED_UP) {
+                        "powder_mining_goblin_eggs"
+                    } else "no_filter"
+                    // BLUE_ONLY enum not explicitly used in comparison, as the only
+                    // case that will block it is HIDE_ALL, which is covered above
+                    "blue" -> return "no_filter"
+                    else -> {
+                        ChatUtils.chat(
```

Comment:
please use `ErrorManager.logErrorStateWithData()` instead. optionally without the stack trace.